### PR TITLE
Fix S.Net.Http.Functional.Tests on AoT

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs [hunk 1]
```diff
@@ -206,6 +206,7 @@ public static IEnumerable<object[]> UseCallback_ValidCertificate_ExpectedValuesD
                 //     MustNotCheck,
                 // }
 
+#if PLATFORM_UNIX
```
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: Why are we adding ifdefs like this.  We usually never do that.

**Comment by davidsh** [comment]
@stephentoub @karelz PTAL.   We shouldn't be needing to add ifdef's like this.

**Comment by karelz** [comment]
Yeah, it looks weird. @alexperovich @danmosemsft any insight why we need to add such ifdefs? Isn't there a better way? Or is it just temporary hack?

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: Right, please use partial classes or utility class that are per platform, and they would have properties like BackendSupportsCustomCertificateHandling implemented in each one.

That's the approach we're trying to use throughout. It's more maintainable than #if.

**Comment by alexperovich** [comment]
@davidsh PTAL

**Comment by davidsh** [comment]
@stephentoub PTAL....This involves the Unix Curl stuff.  Does this look like a clean way of factoring out these Unix dependencies to get the tests building/running for AoT?

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.Unix.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+namespace System.Net.Http.Functional.Tests
+{
+    public partial class HttpClientHandler_ServerCertificates_Test
+    {
+        private static bool ShouldSuppressRevocationException
+        {
+            get
+            {
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                    return false;
```
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.Unix.cs]: nit: we always use braces around things like this.

**Pull request merged**